Microsoft.JSInterop/Microsoft.JSInterop/IJSRuntime.cs

```diff
@@ -0,0 +1,37 @@
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.JSInterop
+{
+	/// <summary>
+	/// Represents an instance of a JavaScript runtime to which calls may be dispatched.
+	/// </summary>
+	public interface IJSRuntime
+	{
+		/// <summary>
+		/// Invokes the specified JavaScript function asynchronously.
+		/// <para>
+		/// <see cref="T:Microsoft.JSInterop.JSRuntime" /> will apply timeouts to this operation based on the value configured in <see cref="P:Microsoft.JSInterop.JSRuntime.DefaultAsyncTimeout" />. To dispatch a call with a different timeout, or no timeout,
+		/// consider using <see cref="M:Microsoft.JSInterop.IJSRuntime.InvokeAsync``1(System.String,System.Threading.CancellationToken,System.Object[])" />.
+		/// </para>
+		/// </summary>
+		/// <typeparam name="TValue">The JSON-serializable return type.</typeparam>
+		/// <param name="identifier">An identifier for the function to invoke. For example, the value <c>"someScope.someFunction"</c> will invoke the function <c>window.someScope.someFunction</c>.</param>
+		/// <param name="args">JSON-serializable arguments.</param>
+		/// <returns>An instance of <typeparamref name="TValue" /> obtained by JSON-deserializing the return value.</returns>
+		ValueTask<TValue> InvokeAsync<TValue>(string identifier, object[] args);
```

Comment:
Should args be `object?[]`? It feels a little too restrictive to require that end-users always provide a value here.